darwin fails with:
```
       > clang \
       >  -O2 -Wl,-s -Wl,-z,relro,-z,now -fpic -pie -D_FORTIFY_SOURCE=2 -fstack-protector-all -std=c99 -pedantic -Wall -Wextra -DNDEBUG \
       >        ynetd.c \
       >      -o ynetd
       > clang: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument]
       > ynetd.c:30:39: warning: a function declaration without a prototype is deprecated in all versions of C [-Wstrict-prototypes]
       >    30 | __attribute__((noreturn)) void version()
       >       |                                       ^
       >       |                                        void
       > 1 warning generated.
       > ld: warning: option -s is obsolete and being ignored
       > ld: unknown option: -z
       > clang: error: linker command failed with exit code 1 (use -v to see invocation)
       > make: *** [Makefile:14: ynetd] Error 1
       For full logs, run:
```

--- a/pkgs/by-name/yn/ynetd/package.nix
+++ b/pkgs/by-name/yn/ynetd/package.nix
@@ -1,0 +1,33 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "ynetd";
+  version = "2024.02.17";
+
+  src = fetchurl {
+    url = "https://yx7.cc/code/ynetd/ynetd-2024.02.17.tar.xz";
+    hash = "sha256-7gioQ0r0LlUftIWKRwTqeZQl0GtskcRKaEE5z6A0S24=";
+  };
+
+  postPatch = lib.optionalString stdenv.isDarwin ''
+    substituteInPlace Makefile --replace-fail "-Wl,-z,relro,-z,now" ""
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    install -Dm755 ynetd $out/bin/ynetd
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "Small server for binding programs to TCP ports";
+    homepage = "https://yx7.cc/code/";
+    license = lib.licenses.mit;
+    platforms = lib.platforms.unix;
+    maintainers = [ lib.maintainers.haylin ];
+    mainProgram = "ynetd";
+  };
+})

